cardinal/ecs/transaction/transaction.go

```diff
@@ -1,11 +1,51 @@
 package transaction
 
 import (
+	"sync"
+
 	"github.com/invopop/jsonschema"
 	"pkg.world.dev/world-engine/sign"
 )
 
-type TxMap map[TypeID][]TxAny
+type TxQueue struct {
```

Comment:
The changes themselves look OK, but not sure I understand this PR:

1. Why is moving the TxQueue from the `ecs` root dir, into an `ecs/transaction` dir cleaner? 
2. Do we have a problem with the `World` object or both the `World` object and the `ecs` package too? Apologies if I missed something but I don't recall what the conflict we found with the `ecs` package was

Reply:
I see problems with both the World object and the ecs package. They are both doing too much. Creating a new package to implement some feature is difficult because it 1) needs to import the ECS package to do anything interesting and 2) needs to be imported BY the ECS package to be used. To satisfy both these requirements we're forced to just implement it in the ECS package. Ideally, ECS will import some new feature package, and that new feature package will import a handful of sub-package so it can do its job.